### DISCUSSION TITLE
[Feature request] XMPP/jabber support for alert notifications

### DISCUSSION THREAD

**Discussion by timtofan** [grafana/grafana#36982] [Alerting]
PagerDuty and Email is already quite nice. Would be awesome to have jabber support as well.

**Comment by genofire** [comment]
Maybe i write a PR in the next month - so sombody could assign me or make a PR at his own

---
If it is really just a file here,
 - like this webhook: https://github.com/grafana/grafana/blob/master/pkg/services/alerting/notifiers/webhook.go
 - and this lib shoud also be easy to use: https://github.com/mattn/go-xmpp

**Comment by bergquist** [comment]
FYI. We are currently not merging new alert notifiers unless there is a really high demand. The maintenance burden of all notifiers is to big. We will soon start working on back end plugins to make it possible to add new notifiers as plugins instead.

**Comment by genofire** [comment]
Thank you FYI ;(
Is there already a concept for this plugins?
I really like to have this feature and would implement it on my fork (and maintain it later)

**Comment by genofire** [comment]
I have build a branch with works already with xmpp. https://github.com/grafana/grafana/compare/master...genofire:alert-xmpp
I have to improve the image url (second message to a file message) the next days.

**Comment by genofire** [comment]
Done - it would be ready, but there is no PR wanted.

**Comment by genofire** [comment]
@bergquist any new information when and how the alert plugin works?

**Comment by gemboot** [comment]
maybe this will help you : 
https://github.com/opthomas-prime/xmpp-webhook

it will indeed be nice to have XMPP/jabber support for alert notifications embedded to Grafana, is there any progress on this ?

**Comment by genofire** [comment]
@gemboot thx remembers my on my dirty try for github and so on: https://dev.sum7.eu/genofire/hook2xmpp - https://github.com/genofire/hook2xmpp - maybe i will help to merge my ideas to @opthomas-prime version

---

I have not heard about, how could i add this function - how to develop a alert plugin. (any news @bergquist ?)
So i think this issue will be here till then.

**Comment by tmsmr** [comment]
@genofire i'm open for any ideas/PR's. "Native" Jabber support in Grafana would be great tho.

**Comment by genofire** [comment]
@bergquist whats up, is there a way to implement xmpp support now? (e.g. by a api for alerts plugins)
Else i will create a PR in the next weeks for a official alert support

**Comment by bergquist** [comment]
Right now we won't merge more notifiers until we support backend plugins.

I suggest you write a small web service that receives Grafana alert notification webhooks and then sends them to Xmpp. Something like https://github.com/jelmer/prometheus-xmpp-alerts

**Comment by immanuelfodor** [comment]
I created a small dockerized microservice written in Python to redirect the Grafana alert notifications to XMPP MUC rooms through the existing webhook integration, maybe you're interested in it: https://github.com/immanuelfodor/xmpp-muc-message-gateway

**Comment by natellium** [comment]
Hey there! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

I will be closing this discussion, but if you’re still interested in this request being implemented please don’t hesitate to open a [feature request](https://github.com/grafana/grafana/issues/new?assignees=&labels=&projects=&template=1-feature_requests.md). Sorry for the inconvenience, and thanks for your contributions!

cc @grafana/alerting-squad
